### ISSUE TITLE
SortIndex starts from 0 instead of 1 for stocktakeBatch and transactionBatch

### ISSUE THREAD

**Issue by Chris-Petty** [openmsupply/mobile#976]
Build Number: Literally all of them I think, though in this case 2.3.0+

Description: Mobile doesn't really use sortIndex, but on desktop it's expected the line numbers start from 1, not 0.

Fix is straight forward in `createRecord.js`.
```
// Correct
sortIndex: requisition.items.length + 1,
// Wrong
sortIndex: transactionItem.transaction ? transactionItem.transaction.numberOfBatches : 0
sortIndex: stocktakeItem.stocktake ? stocktakeItem.stocktake.numberOfBatches : 0,

// This I guess? Assignee figure it out.
sortIndex: stocktakeItem.stocktake ? stocktakeItem.stocktake.numberOfBatches + 1 : 1,
```
Reproducible: Yes

Reproduction Steps: make transactions or stocktakes on mobile, view collector copy on desktop.

Comments:

**Comment by Chris-Petty** [comment]
When a requisition is pruned of lines with 0 quantity, the sortIndex doesn't update. So you can get different ranges and gaps in the sequence. Should it always start at 0 @bijaySussol 
![image](https://user-images.githubusercontent.com/7684221/58064809-b10b3e80-7bd7-11e9-9043-fdb986d8c942.png)

**Comment by diegoevangelisti** [comment]
> When a requisition is pruned of lines with 0 quantity, the sortIndex doesn't update. So you can get different ranges and gaps in the sequence. Should it always start at 1 @bijaySussol
> ![image](https://user-images.githubusercontent.com/7684221/58064809-b10b3e80-7bd7-11e9-9043-fdb986d8c942.png)
> 
> Actually it's not even the same order as mobile:
> ![image](https://user-images.githubusercontent.com/7684221/58064949-2e36b380-7bd8-11e9-883f-7f9847514afd.png)
> 
> It's just what ever order they're added from the master list and in mobile sorted alphabetically on name.

@Chris-Petty @joshxg  just to check if I understood what to change here. If the idea inovolves just generating sortIndex from index [1] to [no.of items], I agree with your solution `sortIndex: stocktakeItem.stocktake ? stocktakeItem.stocktake.numberOfBatches + 1 : 1`.

But if also we need to send it in the same order as in mobile (sorted alphabetically on name),
maybe sortIndex of every involved stocktakeItem can be rearranged after the requisition is finalised to have both lists sorted in a similar way. Please let me know if I am not making my point clear. Thanks!

**Comment by joshxg** [comment]
@diegoevangelisti 

Don't worry about having the same order for mobile and desktop. In desktop, you can save the sorted order, so that next time you open that record, it has the same order - but in mobile this feature is not used (and it, imo, is not needed, just resort the list..!). As long as we are sending correct data back - i.e. starting from 0, not 1, then it is ok

**Comment by diegoevangelisti** [comment]
@joshxg got it, thanks!

**Comment by joshxg** [comment]
Closed by #1668
